js/components/graph/Sidebar.js

```diff
@@ -4,22 +4,32 @@ import Focus from "./Focus";
 
 export default class Sidebar extends React.Component {
   constructor(props) {
+
     super(props);
     this.state = {
       contentHidden: true,
       focusDisabled: false,
       graphActive: 0,
       graphName: "",
-      toggled: false
+      toggled: false,
+      graphSelection: ""
     };
+    this.handleGraphSelection = this.handleGraphSelection.bind(this);
+
   }
 
-  componentWillUpdate(prevProps) { 
+
+  componentWillUpdate(prevProps, nextProps) { console.log(prevProps); console.log(nextProps)
```

Comment:
@stefurac Use `componentDidUpdate`, which takes two arguments `componentDidUpdate(prevProps, prevState)` (at least for our purposes). In the code below you're confusing props vs. state, `graphSelection` is state so you should only be using `prevState` and `this.state`, not `prevProps` and `this.props`.

Reply:
Thank you so much David! That did the trick.